pilot/cmd/pilot-agent/options/security.go

```diff
@@ -107,6 +111,15 @@ func SetupSecurityOptions(proxyConfig *meshconfig.ProxyConfig, secOpt *security.
 		log.Infof("using credential fetcher of %s type in %s trust domain", credFetcherTypeEnv, o.TrustDomain)
 		o.CredFetcher = credFetcher
 	}
+
+	if UseGkeWorkloadCertificate(security.GkeWorkloadCertChainFilePath,
+		security.GkeWorkloadKeyFilePath, security.GkeWorkloadRootCertFilePath) {
+		o.FileMountedCerts = true
```

Comment:
nit: It appears that workload certificates are going to be signed by a root that Istiod does not have in its peer cert verifier. If the pod is going to present its certificate to Istiod for control plane XDS mTLS, Istiod may not recognize it. Could this cause issues?

Reply:
based on offline discussion: this shouldn't be a problem - the agent will not be presenting this cert to Istiod for XDS connection bringup.

Reply:
Test seems to pass but I would have thought this would stop XDS from using proper auth... https://github.com/istio/istio/blob/925e6e61704893fdbaa1e5bdf4b1b86115cdea37/pkg/istio-agent/xds_proxy.go#L654?

I guess tests pass since we never exercise this code path in our integ tests?

Reply:
Can you elaborate on why the quoted code in xds_proxy.go would stop XDS from using proper auth?

Reply:
Because you set FileMountedCerts and if that is set then we don't send a jwt to istiod for xds so it will break.

Reply:
Based on the comment in the following test for "File mounted certs" in https://github.com/istio/istio/blob/925e6e61704893fdbaa1e5bdf4b1b86115cdea37/pkg/istio-agent/agent_test.go#L136, when FileMountedCerts is used, _"mTLS is always used for authentication with
Istiod, never JWT."_ Futhermore,  ISTIO_META_TLS_CLIENT* need to be specified for the file paths of the certificate used to authenticate to istiod.

t.Run("File mounted certs", func(t *testing.T) {
		// User sets FileMountedCerts. They also need to set ISTIO_META_TLS_CLIENT* to specify the
		// file paths. CA communication is disabled. mTLS is always used for authentication with
		// Istiod, never JWT.

Reply:
I agree. And you are setting FileMountedCerts, which means mTLS will be used to Istiod for XDS which is *not* what we want. 

The workload certificate is totally irrelevant to the XDS connection. 

I encourage you to add a test for this, which I believe will fail.

Reply:
A test has been added for this flow in TestAgent.